library/test/src/helpers/concurrency.rs

```diff
@@ -1,18 +1,14 @@
 //! Helper module which helps to determine amount of threads to be used
 //! during tests execution.
-use std::env;
-use std::thread;
+use std::{env, thread};
 
-#[allow(deprecated)]
 pub fn get_concurrency() -> usize {
-    match env::var("RUST_TEST_THREADS") {
-        Ok(s) => {
-            let opt_n: Option<usize> = s.parse().ok();
-            match opt_n {
-                Some(n) if n > 0 => n,
-                _ => panic!("RUST_TEST_THREADS is `{}`, should be a positive integer.", s),
-            }
+    if let Ok(value) = env::var("RUST_TEST_THREADS") {
+        match value.parse().ok() {
+            Some(n) if n > 0 => n,
+            _ => panic!("RUST_TEST_THREADS is `{}`, should be a positive integer.", value),
```

Comment:
The previous version looks a bit easier to read on this part since it specifically mention `usize`.

Also, I wonder why it needs to check `n > 0` here?

I wonder will it be better to parse to `NonZeroUsize` so the guarantees are already there.

```suggestion
        match value.parse::<std::num::NonZeroUsize>().ok() {
            Some(n) => n.get(),
            _ => panic!("RUST_TEST_THREADS is `{}`, should be a positive integer.", value),
```

Reply:
type inference/coercion is helped by the return type, or do you feel it's too far, preferring the type spelled out close to the declaration